core/src/main/java/org/mule/work/ProcessingMuleEventWork.java

```diff
@@ -55,7 +59,7 @@ protected void doRun()
     {
         try
         {
-            this.resultEvent = this.messageProcessor.process(event);
+            this.resultEvent = messageProcessorExecutionTemplate.execute(messageProcessor, event);
```

Comment:
how is this related to this PR purpose? This probably impacts every single mule task.